packages/mux-uploader/src/index.ts

```diff
@@ -0,0 +1,419 @@
+      const file = this.hiddenFileInput?.files && this.hiddenFileInput?.files[0];
+
+      if (file) {
+        this.dispatchEvent(
+          new CustomEvent('mux-drop', {
+            composed: true,
+            bubbles: true,
+            detail: file,
+          })
+        );
+      }
+    });
+  }
+
+  setProgress(percent: number) {
+    if (this.uploadPercentage) this.uploadPercentage.innerHTML = `${Math.floor(percent)}%`;
+    this.setAttribute('aria-valuenow', `${Math.floor(percent)}`);
+    this.setAttribute('aria-valuetext', `${Math.floor(percent)} percent`);
+
+    switch (this.getAttribute('type')) {
+      case TYPES.BAR: {
+        if (this.progressBar) this.progressBar.style.width = `${percent}%`;
+      }
+      case TYPES.RADIAL: {
+        if (this.svgCircle) {
+          // The closer the upload percentage gets to 100%, the closer offset gets to 0.
+          // The closer offset gets to 0, the more we can see the circumference of our circle. (TD).
+          const offset = this.circumference - (percent / 100) * this.circumference;
+
+          this.svgCircle.style.strokeDashoffset = offset.toString();
+        }
+      }
+    }
+  }
+
+  handleUpload(evt: CustomEvent) {
+    const url = this.getAttribute('url');
+
+    if (!url) {
+      if (this.statusMessage) this.statusMessage.innerHTML = 'No url attribute specified -- cannot handleUpload';
+      throw Error('No url attribute specified -- cannot handleUpload');
+    } else {
+      if (this.statusMessage) this.statusMessage.innerHTML = '';
+    }
+
+    if (this.statusMessage) {
+      this.removeAttribute('upload-error');
+      this.statusMessage.innerHTML = '';
+    }
+
+    this.setAttribute('upload-in-progress', '');
+
+    const upload = UpChunk.createUpload({
+      endpoint: url,
+      file: evt.detail,
+    });
+
+    upload.on('error', (err) => {
+      this.setAttribute('upload-error', '');
+
+      if (this.statusMessage && this.uploadPercentage) {
+        this.statusMessage.innerHTML = 'An error has occurred';
+      }
+      throw Error(err.detail.message);
+    });
+
+    upload.on('progress', (progress) => {
+      this.setProgress(progress.detail);
+    });
+
+    upload.on('success', () => {
+      console.log('Upload complete!');
+    });
+  }
+}
+
+type MuxUploaderElementType = typeof MuxUploaderElement;
+declare global {
+  var MuxUploaderElement: MuxUploaderElementType;
+}
+
+/** @TODO Refactor once using `globalThis` polyfills */
+if (!globalThis.customElements.get('mux-uploader')) {
+  globalThis.customElements.define('mux-uploader', MuxUploaderElement);
+  /** @TODO consider externalizing this (breaks standard modularity) */
+  globalThis.MuxUploaderElement = MuxUploaderElement;
+}
+
+if (!globalThis.customElements.get('mux-uploader-drop')) {
+  globalThis.customElements.define('mux-uploader-drop', MuxUploaderDropElement);
+  //@ts-ignore
+  globalThis.MuxUploaderDropElement = MuxUploaderDropElement;
+}
```

Comment:
should this piece be moved into mux-uploader-drop?

Reply:
This might depend on how we want to expose mux-uploader-drop directly to users of this package.

Reply:
Agree this should probably be defined in the mux-uploader-drop module.